rfcs/config-luaurc.md

```diff
@@ -0,0 +1,66 @@
+# Configure analysis via .luaurc
+
+## Summary
+
+Introduces a way to configure type checker and linter using JSON-like .luaurc files
+
+## Motivation
+
+While Luau analysis tools try to provide sensible defaults, it's difficult to establish the rules that work for all code.
+For example, some packages may decide that unused variables aren't interesting; other packages may decide that all files should be using strict typechecking mode.
+
+While it's possible to configure some aspects of analysis behavior using --! comments, it can be cumbersome to replicate this in all files.
+
+## Design
+
+To solve this problem, we are going to introduce support for `.luaurc` files for users of command-line Luau tools.
+For a given .lua file, Luau will search for .luaurc files starting from the folder that the .lua file is in; all files in the ancestry chain will be parsed and their configuration
+applied. When multiple files are used, the file closer to the .lua file overrides the settings.
+
+.luaurc is a JSON file that can also contain comments and trailing commas. The file can have the following keys:
+
+- `"mode"`: type checking mode, can be one of "nocheck", "nonstrict", "strict"
+- `"lint"`: lints to enable; points to an array of string literals that correspond to the names of linting rules (see https://luau-lang.org/lint), or `"*"` that means "all rules"
+- `"nolint"`: lints to disable; points to an array of string literals or `"*"`
+- `"lintErrors"`: a boolean that controls whether lint issues are reported as errors or warnings (off by default)
+- `"typeErrors"`: a boolean that controls whether type issues are reported as errors or warnings (on by default)
```

Comment:
The name of these two makes me think this is an array, but I'm not sure what's a better name for it.

Reply:
I agree :) I also couldn't come up with anything that sounds the same between lint issues and type issues, and is concise and easy to understand. Suggestions welcome!

Reply:
typeErrorSeverity: "warning" | "error"?

Reply:
This could work for type errors but it unfortunately means that for lints we'll need to give up on an easy extensibility route for individual control of errors, see "Alternatives".